browser-extension/mv3/src/service-worker/services/clientHandler.ts

```diff
@@ -74,11 +89,25 @@ export const initClientHandler = async () => {
   console.log("[initClientHandler]");
   const isExtensionStatusEnabled = await isExtensionEnabled();
   setupClientScript(isExtensionStatusEnabled);
+  cacheBlockedDomains();
```

Comment:
Instead of triggering cache ourself, let `getBlockedDomains` handle the caching if it isn't cached

Reply:
`cacheBlockedDomains()` is asynchrounous so if `getBlockedDomains()` is made to handle caching then `getBlockedDomains` has to return a promise. 
It will be better if `getBlockedDomains()` does not return promise because of its usage in `webRequestInterceptor`.

Reply:
If we trigger the caching ourself then there would be no instance when blocked domains have not been cached

Reply:
There can be still be some instance i think if webrequestinterceptor is initialized before this `cacheBlockedDomains` is still in the promise queue. So in this case too, the cachedBlockedDomains isn't ready and webRequestInterceptor bypasses it
But this looks weird and out the place as this method responsibility shouldn't be to initialize the cache. It should be in the concerned file only.

If we initialize it in getBlockedDomains only, there is very small chance that the blockedDomains aren't cached (and that too when extension service worker is init by a website that is in the blocklist).